scripts/o3de/o3de/manifest.py

```diff
@@ -506,7 +540,10 @@ def get_gem_json_data(gem_name: str = None, gem_path: str or pathlib.Path = None
         return None
 
     gem_path = pathlib.Path(gem_path).resolve()
-    gem_json = gem_path / 'gem.json'
+    if gem_path.suffix == '.json':
```

Comment:
Yeah this is definitely a workaround to restore previous behavior which used to allow any file to be passed in to this function and not just gem directories with a `gem` subdirectory https://github.com/o3de/o3de/commit/c778606c89e795ff885f5a732eaecd3b8946ba98#diff-bfe24e34b0c7a56cf773d75fc708e8aebfa786ab6dfc9e94ab93258ff5a6c24aL522

The function used to do this:
```
    if pathlib.Path(gem_path).is_file():
        return get_json_data_file(gem_path, 'gem', validation.valid_o3de_gem_json)
    else:
        return get_json_data('gem', gem_path, validation.valid_o3de_gem_json)
```
There are multiple ways we can address this, one of which is for Project Manager to call `get_json_data_file` directly if you approve of re-adding that method, otherwise we can explore other options.

Reply:
Actually I don't know why that behavior was removed.
Both `get_json_data_file` and `get_json_data` was made to reduce the duplication of code and that change just re-added the copy and paste issues back.
I am OK with adding the previous behavior back.

Reply:
OK, I've restored the previous `get_json_data_file` and `get_json_data` functionality and did some smoke testing like running unit tests, creating a project, downloading remote gems etc.  I also noticed the default user folder functionality got regressed so I restored that also.